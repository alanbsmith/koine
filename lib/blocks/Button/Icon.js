import React from 'react';
import styled from 'styled-components';
import FontAwesome from 'react-fontawesome';
import { omit } from 'lodash';
import { rem } from 'polished';
import { applyStyleModifiers } from 'styled-components-modifiers';

const modifierConfig = {
  left: () => `
    padding-right: ${rem(8)};
  `,
  right: () => `
    padding-left: ${rem(8)};
  `,
};

function UnstyledFontAwesome(props) {
  const iconProps = omit(props, ['modifiers']);
  return <FontAwesome {...iconProps} />;
}

const Icon = styled(UnstyledFontAwesome)`
  ${applyStyleModifiers(modifierConfig)}
`;

export default Icon;
