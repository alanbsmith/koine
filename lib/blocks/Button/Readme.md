```jsx
<FlexContainer>
  <FlexRow>
    <FlexColumn>
      <Button modifiers={['bgPrimary', 'hoverBgPrimary']}>
        <Button.Icon name="home" modifiers={['left']} />
        <Button.Text>action</Button.Text>
      </Button>
    </FlexColumn>
    <FlexColumn>
      <Button>
        <Button.Text>action</Button.Text>
        <Button.Icon name="home" modifiers={['right']} />
      </Button>
    </FlexColumn>
  </FlexRow>
</FlexContainer>
```
