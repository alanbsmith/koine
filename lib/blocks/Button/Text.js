import styled from 'styled-components';
import { rem } from 'polished';

const Text = styled.span`
  font-size: ${rem(12)};
  font-weight: ${props => props.theme.fontWeights.light};
  text-transform: uppercase;
`;

export default Text;
