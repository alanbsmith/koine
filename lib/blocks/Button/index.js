import styled from 'styled-components';
import { lighten, rem } from 'polished';
import { applyStyleModifiers } from 'styled-components-modifiers';

import Icon from './Icon';
import Text from './Text';

import {
  bgColors,
  hoverBgColors,
} from '../../modifiers';

const customModifiers = {
  fullWidth: () => `flex: 1`,
};

const modifierConfig = {
  ...bgColors,
  ...hoverBgColors,
  ...customModifiers,
};

const Button = styled.button`
  align-items: center;
  background-color: ${props => props.theme.colors.base.blue};
  border-radius: 2px;
  border: solid 1px transparent;
  color: ${props => props.theme.colors.base.chrome000};
  display: flex;
  font-size: ${rem(16)};
  justify-content: center;
  line-height: 0;
  padding: ${rem(8)};
  outline: none;
  transition: all 200ms ease;

  &:hover,
  &:focus {
    background-color: ${props => lighten(0.05, props.theme.colors.base.blue)};
    box-shadow: 0 8px 8px 0 ${props => props.theme.colors.base.shadow};
  }

  ${applyStyleModifiers(modifierConfig)}
`;

Button.Icon = Icon;
Button.Text = Text;

export default Button;
