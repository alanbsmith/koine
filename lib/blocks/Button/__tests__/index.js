import React from 'react';
import Button from '../index';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';


describe('Button Block', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Button>
        <Button.Text>Login</Button.Text>
        <Button.Icon name="home" />
      </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the bgSuccess modifier', () => {
    const tree = renderWithTheme(
      <Button modifiers={['bgSuccess']}>
        <Button.Text>Login</Button.Text>
        <Button.Icon name="home" />
      </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the hoverBgSuccess modifier', () => {
    const tree = renderWithTheme(
      <Button modifiers={['hoverBgSuccess']}>
        <Button.Text>Login</Button.Text>
        <Button.Icon name="home" />
      </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the fullWidth modifier', () => {
    const tree = renderWithTheme(
      <Button modifiers={['fullWidth']}>
        <Button.Text>Login</Button.Text>
        <Button.Icon name="home" />
      </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
