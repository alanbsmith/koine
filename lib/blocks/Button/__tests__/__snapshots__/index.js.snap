// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Block adds the bgSuccess modifier 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #1FB6FF;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 0;
  padding: 0.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  background-color: #00a651;
}

.c0:hover,
.c0:focus {
  background-color: #39beff;
  box-shadow: 0 8px 8px 0 rgba(132,146,166,0.4);
}

.c1 {
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c2 {
  font-size: 1rem;
}

<button
  className="c0"
>
  <span
    className="c1"
  >
    Login
  </span>
  <span
    aria-hidden={true}
    className="fa fa-home c2"
  />
</button>
`;

exports[`Button Block adds the hoverBgSuccess modifier 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #1FB6FF;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 0;
  padding: 0.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c0:hover,
.c0:focus {
  background-color: #39beff;
  box-shadow: 0 8px 8px 0 rgba(132,146,166,0.4);
}

.c0:hover,
.c0:focus {
  background-color: #00c05d;
}

.c1 {
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c2 {
  font-size: 1rem;
}

<button
  className="c0"
>
  <span
    className="c1"
  >
    Login
  </span>
  <span
    aria-hidden={true}
    className="fa fa-home c2"
  />
</button>
`;

exports[`Button Block renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #1FB6FF;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 0;
  padding: 0.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c0:hover,
.c0:focus {
  background-color: #39beff;
  box-shadow: 0 8px 8px 0 rgba(132,146,166,0.4);
}

.c1 {
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c2 {
  font-size: 1rem;
}

<button
  className="c0"
>
  <span
    className="c1"
  >
    Login
  </span>
  <span
    aria-hidden={true}
    className="fa fa-home c2"
  />
</button>
`;
